### DISCUSSION TITLE
Tiled Ksampler

### DISCUSSION THREAD

**Discussion by BlenderNeko** [comfyanonymous/ComfyUI#656] [Show and tell]
I made a tiled version of the Ksamplers which you can find [here](https://github.com/BlenderNeko/ComfyUI_TiledKSampler). It's silently been public for a little while now, but it should now be polished enough for use. It has support for masks, control nets, T2IAdaptors, area conditioning/conditioning masks and gligen. Some time in the future I might add support for style models.

**Comment by fractal-fumbler** [comment]
hello and thanks for work
so these Ksamplers are for big images only if to look at uploaded images?

**Reply by BlenderNeko** [reply]
For when the resolution you want to denoise at no longer fits in VRAM. You're essentially trading time and global awareness for VRAM when using this tiled sampler. So if it still fits within VRAM I wouldn't advice using these nodes, but if you want to upscale your images to sizes that normally don't fit inside VRAM using SD you can e.g. use the tiled sampler along with the tiled controlnet to do that.

**Comment by ntdviet** [comment]
It works fine, but can you explain why the padded strategy is still creating 300-400 steps? I don't see much difference (step-wise) compared to the random strategy while it's result is worse (visible seams)

**Reply by BlenderNeko** [reply]
from the readme:
> The padded tiling strategy tries to reduce seams by giving each tile more context of its surroundings through padding. It does this by further dividing each tile into 9 smaller tiles, which are denoised in such a way that a tile is always surrounded by static contex during denoising. This strategy is more prone to seams but because the location of the tiles is static, this strategy is compatible with uni samplers and has no overhead between steps. However the padding makes it so that up to 4 times as many tiles have to be denoised.

There's also a little gif in the readme that tries to show what happens, or if you have previews enabled you can see it happening. Because it essentially visits the same pixel multiple times It takes more steps than other approches. If the results are not suitable for your workflow, use a different tiling strat.

**Comment by 2blackbar** [comment]
HEy peeps how to install the samplers -nodes , i have them as py files but now what ?where they should be ? why this is not explained in basic docs and were forced to do detective work
